### DISCUSSION TITLE
Are there plans to support reactive from CRUD operations?

### DISCUSSION THREAD

**Discussion by rybick** [jOOQ/jOOQ#14172] [Q&A] [answered]
`RowCountQuery` now implements `Publisher`, which allows for executing insert queries in a non-blocking reactive way (which is great!). 
E.g. with reactor:
```
return Mono.from(
    dsl()
        .insertInto(BOOK)
        .columns(BOOK.ID, BOOK.AUTHOR_ID, BOOK.TITLE)
        .values(book.id, book.authorId, book.title)
)
```
but for simple crud operations it is quite verbose. Are there any plans to support reactive for Records or DAOs? e.g. something like this:

```
BookRecord book = ... 
return Mono.from(book.insertAsync())
```
or
```
Book book = ... // pojo
return Mono.from(bookDAO.insertAsync(book))
```

**Comment by lukaseder** [comment, marked as answer]
Sure, the `DAO` is on the roadmap here: https://github.com/jOOQ/jOOQ/issues/5916. I think that `UpdatableRecord` operations are, too, can't seem to find the issue right now.

I'll try to find the issue again next week and report back here.
